src/rebar_git_resource.erl

```diff
@@ -31,11 +31,14 @@ needs_update(Dir, {git, Url, {tag, Tag}}) ->
     ?DEBUG("Comparing git tag ~s with ~s", [Tag, Current1]),
     not ((Current1 =:= Tag) andalso compare_url(Dir, Url));
 needs_update(Dir, {git, Url, {branch, Branch}}) ->
-    {ok, Current} = rebar_utils:sh(?FMT("git symbolic-ref -q --short HEAD", []),
+    %% Fetch remote so we can check if the branch has changed
+    {ok, _} = rebar_utils:sh(?FMT("git fetch", []),
```

Comment:
Should we fetch just the requested branch to avoid fetching more than necessary? Though I guess this form is safer and I'm okay with that.